### ISSUE TITLE
As a content editor, I want to override the order images are displayed for a document so that I can set the images to display in logical order for joins.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/geniza#1011]
## testing notes

In the QA admin site, edit a document with images.
- [x] In the field labeled "image order override", click and drag the images around to reorder them. Confirm this interaction works as expected
- [x] Save the document with a changed order and confirm the new order is persisted in the admin
- [x] View the document in the public site and confirm the order is shown correctly there as well
- [x] Try the same on a document joining multiple fragments each with images, and confirm all of the above still works

---

## dev notes
- [x] add an override image order field to `Document` (array of url fields)
- [x] add custom logic in document admin form to display all thumbnails and add javascript to drag and drop to re-order; when sorted, populate override array field with the canvas ids in order; reset option to revert back to no override
- [x] revise `Document.iiif_images` to use override order when set (should update indexing, document details, and sync transcription)
- [x] revise `DocumentManifestView` to use override order when set (maybe update to use `iiif_images`)

**Comment by kseniaryzhova** [comment]
@rlskoeser it all works for me!

**Comment by richmanrachel** [comment]
@rlskoeser - the re-arranging works, but the text does not adjust to correct page even when I made the cover pages not part of the document in the related fragment section: https://test-geniza.cdh.princeton.edu/en/documents/9117/

[This was the same example I showed yesterday in the meeting where JTS added the cover note about it being the autograph of a famous poet. The transcription is still sticking to the cover pages rather than the manuscript]

**Comment by rlskoeser** [comment]
@richmanrachel text won't adjust until the sync transcription runs again. Do you want me to run it on a specific document or two so you can verify? (probably good to check, the tei migration will use the same logic)

**Comment by richmanrachel** [comment]
@rlskoeser - I see. I just updated the page and it's working! I think we can close - maybe we can just add a note when we release this feature that you will need to wait on the hourly sync for everything to work properly?

**Comment by rlskoeser** [comment]
@richmanrachel ah, thanks for checking back! Glad we set that script to run more frequently. That makes sense to me! 

It would be good if you could think of any high priority documents where you want to have research assistants review and adjust the order before we run the TEI to annotation migration — that will be a one-time migration when we run it for real in production, and we should honor any overrides that are set, but if they are not set then you'll have to override order and reassociate content after the migration.

**Comment by richmanrachel** [comment]
@rlskoeser - hmmm... that's a good idea, though I probably can only trust the most senior research assistants to do that work (Alan, Amir, and Moshe). I think our transcription cleanup log (2nd page) is still the best place to start with the ones that are incomplete: https://docs.google.com/spreadsheets/d/1S8eorROnapeEl3Xlqk352SFwrLkuHbTgL69r94IMo88/edit#gid=286636463

What is the timing of fixing this?

**Comment by kseniaryzhova** [comment]
@rlskoeser not an issue per se, but the dragging for reaerranging the images doesn't seem intuitive to me - if you drag an image too far outside the confines of another image then you get this red crossed out circle and the image will not rearrange. These thumbnails are small though so it's not a lot of space to work with (and in most situations where I've had to drag an image/file there's more space to play with). I did eventually figure it out though and it does work as intended (which is why I hesitated to write this up).

**Comment by rlskoeser** [comment]
@kseniaryzhova thanks for documenting, that's good for us to know

**Comment by rlskoeser** [comment]
Closing per discussion at meeting yesterday — current version is acceptable. Ksenia's problems were on MS Edge; feature worked fine for RR and RSK on Firefox and Chrome. If researchers have difficulty using it we can revisit/refine later.
